File: clojure/armstrong-numbers/src/armstrong_numbers.clj
(ns armstrong-numbers)

(require '[clojure.math.numeric-tower :as mat])

(defn armstrong? [num]
  (= num (loop [num-sum 0
                num-pos 0
                num-len (count (str num))
                num-seq (map #(- % 48) (map int (seq (char-array (str num)))))]
           (if (= num-pos num-len) num-sum
               (recur (+ num-sum (mat/expt (first num-seq) num-len))
                      (inc num-pos)
                      num-len
                      (rest num-seq))))))

File: typescript/armstrong-numbers/armstrong-numbers.ts
class ArmstrongNumbers {
    static isArmstrongNumber(num: number) {
        let armnum = ((x) => {
            let xarr = Array.from(x.toString());
            let sum = 0;
            for (let pos = 1; pos <= xarr.length; pos++) {
                sum += (parseInt(xarr[pos-1]) ** (xarr.length));
            }
            return sum;
        })(num);

        return (num == armnum);
    }
}

export default ArmstrongNumbers;